olive/engine/engine.py

```diff
@@ -668,9 +675,19 @@ def _run_passes(
                 model.model_storage_kind == ModelStorageKind.AzureMLModel
                 and not self.host_for_pass(pass_id).system_type == SystemType.AzureML
             ):
-                error_msg = "Azure ML model only supports AzureMLSystem for Olive Pass"
-                logger.error(error_msg)
-                raise Exception(error_msg)
+                if not self.azureml_client_config:
+                    raise ValueError("AzureML client config is required to download the model from AzureML storage")
+                model_download_path = self._model_cache_path / "azureml_input_model"
+                model_path = model.download_from_azureml(
+                    self.azureml_client_config.create_client(), model_download_path
```

Comment:
better to move out of the for loop?

Reply:
Since each pass and evaluator can have its own host and target, I was thinking one case that all previous passes and evaluations are running on AML but one of the middle are running locally. This is to handle that case so I put this in the loop.

Reply:
Although currently, we will need to downlaod the model from the AML after the first pass run on AML, this is a gap from AML side but not our side. We will support this logic in the future once AML fixes this.